### ISSUE TITLE
Daml Assistant's project root option produces strange behaviour

### ISSUE THREAD

**Issue by gaborh-da** [digital-asset/daml#2449]
```
~/work/lib-daml-jvm-test$ daml build --project-root src/test/resources/ping-pong/ --output src/test/resources/ping-pong.dar
damlc: /home/gabre/work/lib-daml-jvm-test/src/test/resources/ping-pong/src/test/resources/ping-pong: changeWorkingDirectory: does not exist (No such file or directory)
```

Note that `src/test/resources/ping-pong` is an existing directory with a daml.yaml:
```
~/work/lib-daml-jvm-test$ ls src/test/resources/ping-pong/
daml  daml.yaml
```

Other existing directories with or without a daml.yaml produce the same behaviour:
```
~/work/lib-daml-jvm-test$ daml build --project-root scripts/ --output src/test/resources/ping-pong.dar
damlc: /home/gabre/work/lib-daml-jvm-test/scripts/scripts: changeWorkingDirectory: does not exist (No such file or directory)
```

Note the "duplication" of the given relative project root path, e.g. *lib-daml-jvm-test/scripts/scripts* while I gave it simply *scripts*.

**SDK Version: 0.13.18**

**Comment by cocreature** [comment]
The issue is caused by the fact that `damlc build` calls `damlc init` and both use `withProjectRoot`. So the `withProjectRoot` call in `damlc build` ends up changing to the right directory but then `damlc init` tries to change again misinterpreting the relative project root as relative to the changed working directory. Shouldn’t be too hard to fix.

As a workaround, you can either use an absolute project root, e.g. `$PWD/src` or you can run `damlc init --project-root …` and then `damlc build --project-root … --init-package-db=no`.

**Comment by cocreature** [comment]
Fixed in #2459
